### DISCUSSION TITLE
Port 10000 is dropped after setting up Budibase with Docker

### DISCUSSION THREAD

**Discussion by Grayda** [Budibase/budibase#8660] [Help]
I'm using the [Portainer app template](https://docs.budibase.com/docs/portainer) setup to run Budibase on my local network, and after adding the stack and attempting to go to https://my-server-ip:10000, the port gets dropped and I'm redirected to https://my-server-ip/builder

Last time I installed Budibase locally (which would have been about a year ago) I needed to edit `envoy.yaml` to get things running, but the installation instructions don't mention this -- it sems like all you should have to do is add the template, fill out the basics (i.e. JWT secret etc.) and in a minute or two you're ready to hit port 10000 and start making apps. 

So am I missing something in the setup? Or Should this be everything I need to do?

**Comment by melohagan** [comment]
Hey @Grayda 

Envoy is no longer used. It is now nginx: https://docs.budibase.com/docs/reverse-proxy

I think by default you shouldn't need to do anything after installation.

**Reply by Grayda** [reply]
I haven't touched anything post-installation. Here's my exact steps:

1. I added Budibase as an app template as per these steps: https://docs.budibase.com/docs/portainer
2. I went to App Templates > Clicked on Budibase
3. I left all of the settings unchanged. Even the keys / passwords / secrets (though it did the same thing when I changed them)
4. When the stack deploys successfully, I go to http://my-ip-address:10000 and the browser redirects to http://my-ip-address/builder . In other words, the port is not maintained when redirecting. 

The instructions don't say to do anything else, but do I need to update any nginx configurations? If I do, those should be documented too.

EDIT: Interestingly, if I go to http://my-ip-address:10000, the port gets dropped. However if I go to http://my-ip-address:10000/builder/, then the port stays and I can actually get to Budibase.

So it looks like the initial redirect to /builder doesn't maintain the port. 

EDIT 2: Looks like the Portainer app template doesn't expose the `BB_ADMIN_USER_EMAIL` and `BB_ADMIN_USER_PASSWORD` variables, so I wasn't sure how to log in after the stack had deployed. In the end I manually created a stack, using `docker-compose.yaml` and `.env` from the docker compose instructions, then I was able to set a username and password and log in.

But on top of all of this, the self-hosted version seems to be out of date (?). If I create an app via Budibase cloud, the app is on version 2.1.25, but if I create an app on the self-hosted version, the app is on version  1.0.59 which is apparently the latest version. I've updated the `docker-compose.yaml` to use `budibase/<image name here>` instead of `budibase.docker.scarf.sh/budibase/<image name here>`, ~~but now I just get a 502 bad gateway error.~~ and that seems to have worked, as the interface looks the same as the cloud version. 

~~I really want to use Budibase with a MySQL server on my local network (because the inbuilt database doesn't support the features I need), but keep running into little but frustrating things that make it tough to get started.~~

**Reply by Grayda** [reply]
Okay, so I think I worked out what's going on.

A while ago I installed Budibase. I used it, decided to use the cloud version instead, and deleted the stack and ticked the option to remove the volumes.

But I guess it didn't actually delete the volumes because when I got in to the UI this time around, I noticed my old test app was there, but I wasn't able to delete it because it said the app wasn't found. So I went into the Volumes tab in Portainer and manually deleted the volumes from there, then re-deployed the stack. Now I can hit http://my-ip-address:10000 and it will redirect to /builder/ AND keep the port.

So yeah, I got the issue sorted.. after having to manually clear some data from Portainer. 

EDIT: And I now realize that the MySQL connector doesn't support attachments. Looks like I'll need to rethink my app because I wanted a dynamic filter that works on a relationship, but I also want to be able to attach images. There's some workarounds for that that I can use, so I'll look at that. Oh well, still easier than building an app from scratch using Laravel like I did before :P
